### DISCUSSION TITLE
QSO data mismatch hardware interface data

### DISCUSSION THREAD

**Discussion by R1BLH** [wavelog/wavelog#426] [General]
Sometimes i found a strange situation. 
I'm using FT-857 controlled by PC via Omnirig. To connect Omnirig with Wavelog I used CloudCat application.
Information about the transceiver frequency is normally displayed on the dashboard, but to the QSO entry form only the band information is transferred correctly, but the frequency field is filled with the default frequency for the band.
Please have a look at the screenshots:
Omnirig (CloudCat):
![изображение](https://github.com/wavelog/wavelog/assets/16667173/d6349886-1948-4b0d-8469-57c217039a0f)
Dashboard:
![изображение](https://github.com/wavelog/wavelog/assets/16667173/2a70ce4c-27dc-4b8e-95c3-b36fd877a24e)
Station data from new QSO form:
![изображение](https://github.com/wavelog/wavelog/assets/16667173/62432578-d75e-41dc-a6f1-64e29a984c9c)
QSO form data:
![изображение](https://github.com/wavelog/wavelog/assets/16667173/24f53fde-db14-45f0-92bc-194fd62800aa)

Update:
Before transmit all was OK.
During transmit all is 0:
![изображение](https://github.com/wavelog/wavelog/assets/16667173/555065c4-cbda-484c-8f93-a5abb23fdbb9)
After switching from transmit to receive band is correct, frequency is default:
![изображение](https://github.com/wavelog/wavelog/assets/16667173/dab5137a-95ab-4382-9347-9a62ba1de150)

**Comment by HB9HIL** [comment]
Hello @R1BLH

There is indeed a bug. Gonna fix it.

Created issue in #430 
Fix in #431 

@R1BLH 

The issue with Frequency = 0 must come from your CAT Interface. This has nothing to do with Wavelog.

**Reply by R1BLH** [reply]
> The issue with Frequency = 0 must come from your CAT Interface. This has nothing to do with Wavelog.

Agreed. It's a onmirigs bug. :) Picture shows this.

**Comment by HB9HIL** [comment]
@R1BLH bugfix is in `dev`branch. This should fix it. Thanks for reporting

**Reply by R1BLH** [reply]
@HB9HIL Found a strange behaviour when using Omnirig (via CloudlogCat).
When trying to save a QSO, the message is displayed that the frequency (range) value is not filled.
This is because during transmission (as described above) a null frequency/band is written to the form. As a result, after the end of the QSO ( switching to receive) it takes some time for the data to be updated from the hardware interface, and only after the update can the QSO be saved.
Is it possible to add a frequency (and band) equal to zero to the exceptions? I mean, not to enter this value in the form/display, but to use it as the true previous frequency value. A frequency equal to zero is obviously a false value.

**Reply by HB9HIL** [reply]
Did you try with [WLgate](https://github.com/wavelog/WaveLogGate)?

**Reply by R1BLH** [reply]
No. I not use FLRig. But FLRig is prerequisite to the WLgate.

I'll try use it.

**Reply by HB9HIL** [reply]
The cool thing about WLgate is that you just need one application for everything. WLgate can push QSO from WSJT-X and can push QRG/Mode from FLrig to Wavelog. No need for two programms. In addition FLrig has some really cool features you can use.

**Reply by R1BLH** [reply]
Yes, I have tried a bunch of WLGate and FLRig.
It is not very convenient that the UDP port is not configurable in WLGate and there is no second server to which the messages received from JTDX are sent. I am used to using Gridtracker for logging and visualization. 
It is possible to change the order of mapping so that messages from JTDX do not reach WLGate. 

But there is a problem: there is a delay when enabling transmissions from JTDX. It may be possible to fix this with FLRig settings.

**Reply by HB9HIL** [reply]
Start a Discussion here :)

https://github.com/wavelog/wavelog/discussions/categories/wl-gate
